Framework/Algorithms/src/ConvertUnits.cpp

```diff
@@ -682,21 +678,30 @@ API::MatrixWorkspace_sptr ConvertUnits::removeUnphysicalBins(
                                                  maxBins - 1);
     // Next, loop again copying in the correct range for each spectrum
     for (int64_t j = 0; j < int64_t(numSpec); ++j) {
-      int k;
-      for (k = 0; k < lastBins[j] - 1; ++k) {
-        result->mutableX(j)[k] = workspace->x(j)[k];
-        result->mutableY(j)[k] = workspace->y(j)[k];
-        result->mutableE(j)[k] = workspace->e(j)[k];
-      }
-      result->mutableX(j)[k] = workspace->x(j)[k];
-      ++k;
-      // If necessary, add on some fake values to the end of the X array (Y&E
-      // will be zero)
+      auto edges = workspace->binEdges(j);
+      auto &resX = result->mutableX(j);
+
+      auto k = lastBins[j];
+
+      std::copy(edges.cbegin(), edges.cbegin() + lastBins[j],
+                result->mutableX(j).begin());
+
+	  // If necessary, add on some fake values to the end of the X array (Y&E
+	  // will be zero)
       if (k < maxBins) {
```

Comment:
I think you can drop the `if`. It is implicit by the range that you give to `std::transform`, I think?